src/sentry/grouping/component.py

```diff
@@ -37,44 +41,35 @@ class GroupingComponent:
 
     def __init__(
         self,
-        id,
-        hint=None,
-        contributes=None,
-        values=None,
-        variant_provider=False,
-        tree_label=None,
-        is_prefix_frame=None,
-        is_sentinel_frame=None,
+        id: str,
+        hint: str | None = None,
+        contributes: bool | None = False,
+        values: Sequence[str | GroupingComponent] | None = None,
+        variant_provider: bool = False,
+        tree_label: str | None = None,
+        is_prefix_frame: bool | None = None,
+        is_sentinel_frame: bool | None = None,
     ):
         self.id = id
 
         # Default values
-        self.hint = DEFAULT_HINTS.get(id)
-        self.contributes = None
+        self.hint = hint
```

Comment:
to keep it exactly as it worked before, it would need to be 
```suggestion
        self.hint = hint or DEFAULT_HINTS.get(id)
```
Not sure if that default value is actually needed though.